ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java

```diff
@@ -412,7 +419,6 @@ protected XARecoveryModule(XARecoveryResourceManager recoveryClass, String logNa
         _logName = logName;
         _recoveryManagerClass = recoveryClass;
         if (_recoveryManagerClass == null) {
-            this.setRecoveryProblems(true);
```

Comment:
Please can you explain why this is no longer classed as a recovery problem?

Reply:
IMO, this is not a problem during recovery but rather a problem during the initialisation of Narayana.

Reply:
Plus, as `setRecoveryProblems(true)` is invoked in the constructor, as soon as `periodicWorkFirstPass()` is invoked from `PeriodicRecovery`, `recoveryProblems` will be set to false (i.e. `setRecoveryProblems(false)`)

Reply:
I think that if is null then we will eventually end up in https://github.com/jbosstm/narayana/blob/main/ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java#L211 because of a NullPointerException at https://github.com/jbosstm/narayana/blob/main/ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java#L198 so that would make `setRecoveryProblems(true)`.

I am not super sure I agree that an invalid configuration should not be considered a problem for recovery but we can think more on this.

Reply:
IMO, `recoveryProblem` shouldn't be used to report problems during initialisation (at least, as far as I can tell from other use cases). Moreover, before we end up at line [211](https://github.com/jbosstm/narayana/blob/main/ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java#L211), we hit [setRecoveryProblems(false)](https://github.com/jbosstm/narayana/blob/main/ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java#L188), which makes setting `recoveryProblems` to true during initialisation pointless

Reply:
I think we should be safe